test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -892,6 +892,185 @@ UCS_TEST_P(test_ucp_mmap, fixed) {
+
+        for (int i = 0; i < count; i++) {
+            chunk.push_back(mem_chunk::create(sender().ucph()));
+            EXPECT_NE(NULL, uintptr_t(chunk.back().memh));
+        }
+    }
+
+    void cleanup() override
+    {
+        for (auto &c : chunk) {
+            c.destroy();
+        }
+
+        test_ucp_mmap::cleanup();
+    }
+};
+
+ucp_rkey_h test_ucp_rkey_compare::mem_chunk::unpack(ucp_ep_h ep)
+{
+    ucs_status_t status;
+    ucp_rkey_h rkey;
+
+    status = ucp_ep_rkey_unpack(ep, rkey_buffer, &rkey);
+    ASSERT_UCS_OK(status);
+    rkeys.push_back(rkey);
+    return rkey;
+}
+
+void test_ucp_rkey_compare::mem_chunk::destroy()
+{
+    ucs_status_t status;
+
+    if (memh != NULL) {
+        for (auto &rkey : rkeys) {
+            ucp_rkey_destroy(rkey);
+        }
+
+        ucp_rkey_buffer_release(rkey_buffer);
+        status = ucp_mem_unmap(context, memh);
+        ASSERT_UCS_OK(status);
+        memh = NULL;
+    }
+}
+
+test_ucp_rkey_compare::mem_chunk
+test_ucp_rkey_compare::mem_chunk::create(ucp_context_h context)
+{
+    void *rkey_buffer           = NULL;
+    size_t size                 = 4096;
+    ucp_mem_map_params_t params = {
+        .field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                      UCP_MEM_MAP_PARAM_FIELD_LENGTH  |
+                      UCP_MEM_MAP_PARAM_FIELD_FLAGS,
+        .address    = NULL,
+        .length     = size,
+        .flags      = UCP_MEM_MAP_ALLOCATE,
+    };
+    ucs_status_t status;
+    ucp_mem_h memh;
+    size_t rkey_size;
+
+    status = ucp_mem_map(context, &params, &memh);
+    if (status != UCS_OK) {
+        memh = NULL;
+        goto out;
+    }
+
+    status = ucp_rkey_pack(context, memh, &rkey_buffer, &rkey_size);
+    if (status != UCS_OK) {
+        status = ucp_mem_unmap(context, memh);
+        ASSERT_UCS_OK(status);
+        memh = NULL;
+        goto out;
+    }
+
+out:
+    return {context, memh, rkey_buffer};
+}
+
+UCS_TEST_P(test_ucp_rkey_compare, rkey_compare_errors)
+{
+    ucp_rkey_compare_params_t params = {};
+    ucp_rkey_h rkey                  = chunk.front().unpack(receiver().ep());
+    ucs_status_t status;
+    int result;
+
+    status = ucp_rkey_compare(receiver().ucph(), rkey, rkey, &params, NULL);
+    ASSERT_EQ(UCS_ERR_INVALID_PARAM, status);
+    params.field_mask = 1;
+    status = ucp_rkey_compare(receiver().ucph(), rkey, rkey, &params, &result);
+    ASSERT_EQ(UCS_ERR_INVALID_PARAM, status);
+}
+
+UCS_TEST_P(test_ucp_rkey_compare, rkey_compare_different_config)
+{
+    int result                       = 1;
+    ucp_rkey_compare_params_t params = {};
+    ucp_rkey_h rkey1, rkey2;
+    ucs_status_t status;
+    ucp_worker_cfg_index_t *cfg_index;
+
+    rkey1 = chunk[0].unpack(receiver().ep());
+    rkey2 = chunk[0].unpack(receiver().ep());
+
+    if (m_ucp_config->ctx.proto_enable) {
+        cfg_index = &rkey1->cfg_index;
+    } else {
+        cfg_index = &rkey1->cache.ep_cfg_index;
+    }
+
+    (*cfg_index)++;
```

Comment:
can we do this by unpacking with a different md_map, for example?
use ucp_ep_rkey_unpack_internal

Reply:
fixed, generating new config by using unpack internal instead.